src/main/java/io/vertx/core/spi/cluster/ClusterManager.java

```diff
@@ -83,20 +75,56 @@
    */
   void nodeListener(NodeListener listener);
 
+  /**
+   * Store the details about this clustered node.
+   */
+  Future<Void> setNodeInfo(NodeInfo nodeInfo);
```

Comment:
instead pass `Promise<Void>` so the `ClusterManager` does not need to care about which context to use

Reply:
done